Doc/c-api/init.rst

```diff
@@ -1074,8 +1078,11 @@ with sub-interpreters:
    Ensure that the current thread is ready to call the Python C API regardless
    of the current state of Python, or of the global interpreter lock. This may
    be called as many times as desired by a thread as long as each call is
-   matched with a call to :c:func:`PyGILState_Release`. In general, other
-   thread-related APIs may be used between :c:func:`PyGILState_Ensure` and
+   matched with a call to :c:func:`PyGILState_Release`, *except* when the
+   thread creates a sub-interpreter with it's own :term:`GIL` (see :c:member:`PyInterpreterConfig.gil`).
+   In that case, the call to :c:func:`PyGILState_Release` should be replaced with :c:func:`Py_EndInterpreter`.
```

Comment:
```suggestion
   In that case, :c:func:`PyGILState_Release` should be used instead of
   :c:func:`Py_EndInterpreter` to release that interpreter's GIL.
```

Reply:
I should read suggestions before committing them :sweat_smile:

`Py_EndInterpreter` and `PyGILState_Release` should be swapped here. If the thread creates a subinterpreter, `Py_EndInterpreter` will release the GIL. If you call `PyGILState_Release` instead, then you no longer have the GIL, and can't end the subinterpreter.

Reply:
Ah no it's my bad, sorry :(